Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/baseprograms/BaseTeleOp.java

```diff
@@ -0,0 +1,210 @@
+
+            // Code added to draw the pose, remove before competition, causes lags:
+            TelemetryPacket p = new TelemetryPacket();
+            Canvas c = p.fieldOverlay();
+            c.setStroke("#3F5100");
+            MecanumDrive.drawRobot(c, drive.pose);
+
+            FtcDashboard dashboard = FtcDashboard.getInstance();
+            dashboard.sendTelemetryPacket(p);
+
+            if (armMotor != null && dumperServo != null && gateServo != null) {
+                outputUsingControllers();
+                intakeUsingControllers();
+                telemetry.addData("ArmMotor", armMotor.getCurrentPosition());
+                telemetry.addData("DumperServo", dumperServo.getPosition());
+                telemetry.addData("GateServo", gateServo.getPosition());
+            }
+
+            telemetry.addData("FRMotor", FR.getCurrentPosition());
+            telemetry.addData("FRMotor", FR.getPowerFloat());
+            telemetry.addData("FLMotor", FL.getCurrentPosition());
+            telemetry.addData("FLMotor", FL.getPowerFloat());
+            telemetry.addData("BRMotor", BR.getCurrentPosition());
+            telemetry.addData("BRMotor", BR.getPowerFloat());
+            telemetry.addData("BLMotor", BL.getCurrentPosition());
+            telemetry.addData("BLMotor", BL.getPowerFloat());
+            telemetry.update();
+        }
+
+        // Close drive (release resources)
+        drive.close();
+    }
+
+    boolean leftBumperIsPressed = false;
+
+    public void resetIMUIfNeeded() {
+        if (gamepad1.left_bumper && !leftBumperIsPressed) {
+            IMU.Parameters parameters;
+            if (drive.isDevBot) {
+                parameters = new IMU.Parameters(new RevHubOrientationOnRobot(
+                        RevHubOrientationOnRobot.LogoFacingDirection.UP,
+                        RevHubOrientationOnRobot.UsbFacingDirection.FORWARD));
+            } else {
+                parameters = new IMU.Parameters(new RevHubOrientationOnRobot(
+                        RevHubOrientationOnRobot.LogoFacingDirection.RIGHT,
+                        RevHubOrientationOnRobot.UsbFacingDirection.UP)); }
+            drive.imu.initialize(parameters);
+        }
+        leftBumperIsPressed = gamepad1.left_bumper;
+    }
+
+    public boolean sensitive = false;
+
+    public void driveUsingControllers() {
+        sensitive = gamepad1.right_bumper;
+
+        double sensitivity, rotSensitivity;
+        double strafeConstant = 1.1;
+
+        if (sensitive) {
+            sensitivity = 0.5;
+            rotSensitivity = 0.8;
+        } else {
+            sensitivity = 1;
+            rotSensitivity = 1;
+        }
+
+        double x = curveStick(gamepad1.left_stick_x) * strafeConstant * sensitivity;
+        double y = curveStick(-gamepad1.left_stick_y) * sensitivity;
+        double rot = curveStick(gamepad1.right_stick_x) * rotSensitivity;
+
+        mecanumDrive(x, y, rot);
+    }
```

Comment:
Duplicate code is almost always bad. Again, someone make a change to this code and not realize that the almost identical routine below also needs a fix. Alarm bells should always go off when you see duplicate code. In this case, driveUsingControllers() could simply call driveUsingControllers(false) and thereby still support the option of callers using either driveUsingControllers() or driveUsingControllers(boolean).